Storage/Storage/Tools/StorageType+Deletions.swift

```diff
@@ -54,4 +54,29 @@ public extension StorageType {
             deleteObject(category)
         }
     }
+
+    /// Deletes all of the stored Product Tags that don't have an active product relationship
+    ///
+    func deleteUnusedProductTags(siteID: Int64) {
+        let tagsWithNoAssociatedProducts = loadProductTags(siteID: siteID).filter { tag in
```

Comment:
nit: probably can make this one line like `loadProductTags(siteID: siteID).filter { $0.products?.isEmpty == false }` ?

Reply:
I think you indented `== true` 😃  edited 👍

Reply:
actually, if `$0.products` is nil, this is going to be false. maybe `isNilOrEmpty`?

Reply:
Fixed here 9c5c6a5139bcc56590ef8533231b9680489bf651